enterprise/cluster/src/main/java/org/neo4j/cluster/ClusterSettings.java

```diff
@@ -69,8 +71,22 @@ public String toString()
         }
     };
 
+    public enum Mode
+    {
+        SINGLE,
+        HA,
+        ARBITER,
+        CORE,
+        READ_REPLICA
+    }
+
+    @Description( "Configure the operating mode of the database -- 'SINGLE' for stand-alone operation, 'HA' for " +
+            "operating as a member in a cluster, 'ARBITER' for an HA-only cluster member with no database, " +
+            "CORE for a core member of a Causal Clustering cluster, or READ_REPLICA for read replica." )
+    public static final Setting<Mode> mode = setting( "dbms.mode", options( Mode.class ), Mode.SINGLE.name() );
```

Comment:
This seems like a strange place to define this (since it crosses HA and core-edge. Where was it before?